### PR TITLE
Vue: Cleanup Vue Code

### PR DESCRIPTION
**Pull request by ErikCH**
#### Description of changes

- [x] Moved from Jest to Vitest for Vue testing

- [x] Added in JSX - For future tests or primitives
- [x] Fixed issue with Code Editor error Default export of the module has or is using private name in any script setup file that has an interface

#### Issue #, if available
#2027 

#### Description of how you validated changes
Test passed

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are updated
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [x] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/vue/tsconfig.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
     "strict": true,
     "jsx": "preserve",
     "importHelpers": true,
-    "declaration": true,
+    "declaration": false,
```
Comment (ErikCH) [inline packages/vue/tsconfig.json:8]: This fixes the issue with `script setup` and having an `interface` in it, and getting an error message.

File: packages/vue/vite.config.ts [hunk 2]
```diff
@@ -1,17 +1,34 @@
-// vite.config.js
-
+/// <reference types="vitest" />
 import { defineConfig } from 'vite';
 import dynamicImportVars from '@rollup/plugin-dynamic-import-vars';
 import path from 'path';
 import vue from '@vitejs/plugin-vue';
 import typescript2 from 'rollup-plugin-typescript2';
 import Components from 'unplugin-vue-components/vite';
+import vueJsx from '@vitejs/plugin-vue-jsx';
```
Comment (ErikCH) [inline packages/vue/vite.config.ts:8]: We can use JSX in our tests now, or in our components if needed.

File: packages/vue/package.json [hunk 3]
```diff
@@ -47,25 +47,29 @@
...
+    "@types/node": "^17.0.43",
+    "@vitejs/plugin-vue": "^2.3.3",
+    "@vitejs/plugin-vue-jsx": "^1.3.10",
     "@vue/cli-plugin-unit-jest": "5.0.0-beta.2",
     "@vue/eslint-config-prettier": "^7.0.0",
     "@vue/eslint-config-typescript": "^10.0.0",
-    "@vue/test-utils": "^2.0.0-rc.12",
+    "@vue/test-utils": "^2.0.1",
     "babel-jest": "^26.3.1",
+    "c8": "^7.11.3",
```
Comment (ErikCH) [inline packages/vue/package.json:58]: This is what vitest uses for code coverage.

File: packages/vue/__tests__/Authenticator.spec.tsx [hunk 4]
```diff
@@ -2,7 +2,7 @@ import { mount } from '@vue/test-utils';
 import { Authenticator } from '../dist';
 import crypto from 'crypto';
 
-// This is needed for jest to get crypto module needed by nanoid
+// This is needed for vitest to get crypto module needed by nanoid
 Object.defineProperty(global.self, 'crypto', {
   value: {
     getRandomValues: (arr: Array<any>) => crypto.randomBytes(arr.length),
```
Comment (wlee221) [inline packages/vue/__tests__/Authenticator.spec.tsx:8]: I've been seeing 

image

(and this is on me bc I wrote the code)

Comment (wlee221) [reply packages/vue/__tests__/Authenticator.spec.tsx:8]: Can you check if this polyfill is still needed, and if so, can you try more scoped type for `Array<any>`?

Comment (ErikCH) [reply packages/vue/__tests__/Authenticator.spec.tsx:8]: Yeah I can update that so GH actions does not complain

**Review by wlee221 — APPROVED** [review]
lgtm!

**Review by 0618 — APPROVED** [review]

**Comment by ErikCH** [comment]
Closed for #2184

**Pull request closed without merging**